src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -38,8 +38,8 @@
 
 namespace ripple {
 
-template<class TIn, class TOut, template <class In, class Out> class Policy>
-class BookStep : public StepImp<TIn, TOut, BookStep<TIn, TOut, Policy>>
+template<class TIn, class TOut, class TDerived>
+class BookStep : public StepImp<TIn, TOut, BookStep<TIn, TOut, TDerived>>
 {
 private:
```

Comment:
`protected` instead of `private` and we no longer need to pass in several pieces of state to `qualityUpperBound`.